Document/0x04e-Testing-Authentication-and-Session-Management.md

```diff
@@ -250,7 +250,8 @@ The purpose of this test case is verifying logout functionality and determining
 
 Failing to destroy the server-side session is one of the most common logout functionality implementation errors. This error keeps the session or token alive, even after the user logs out of the application. An attacker who gets valid authentication information can continue to use it and hijack a user's account.
 
-Many mobile apps don't automatically log users out because it is inconvenient for customers by implementing stateless authentication. The application should still have a logout function, and it should be implemented according to best practices, destroying the access and refresh token on the client and server. Otherwise, authentication can be bypassed when the refresh token is not invalidated.
+Many mobile apps don't automatically log users out. There can be various reasons, such as: because it is inconvenient for customers, or because of decisions made when implementing stateless authentication. The application should still have a logout function, and it should be implemented according to best practices, destroying all locally stored tokens or session identifiers. If session information is stored on the server, it should also be destroyed by sending a logout request to that server. In case of a high risk application, tokens should be blacklisted. Not removing tokens or session identifiers can result in unauthorized access to the application in case the tokens are leaked.
```

Comment:
The term "high-risk application" appears several times in the guide and 0 times in the MASVS. It always draws my attention as I know what it is meant but I think at least we should describe it once, in the MASVS? Are we talking about "high-risk applications" when talking about L2? L2+R? What are low- or medium- risk apps for the MASVS? 

BTW. Could you please take the chance to correct "high risk application" in the rest of the guide to "high-risk application"? It appears in both versions (with and without hyphen) and I'd say with hyphen would be correct as it turns into an adjective.

```suggestion
Many mobile apps don't automatically log users out. There can be various reasons, such as: because it is inconvenient for customers, or because of decisions made when implementing stateless authentication. The application should still have a logout function, and it should be implemented according to best practices, destroying all locally stored tokens or session identifiers. If session information is stored on the server, it should also be destroyed by sending a logout request to that server. In case of a high-risk application, tokens should be blacklisted. Not removing tokens or session identifiers can result in unauthorized access to the application in case the tokens are leaked.
```

Reply:
replaced with high-risk application. High-risk could be anything from L2 and/or R, depending on what a devleoper wants. high-risk is hard to explain, as it is all dependingon the threatmodel: if you have a large cyber-security insurance, then it makes no sense to call a medical nuclear controling app high-risk :P as the insurance covers it ^^.